<?xml version="1.0"?>
<launch>
    <!-- Load parameters -->
    <rosparam command="load" file="$(find navigation)/param/ego_planner.yaml" />
    <rosparam command="load" file="$(find navigation)/param/trajectory_planner.yaml" />

    <!-- Start global trajectory planner node -->
    <node pkg="navigation" type="trajectory_planner" name="trajectory_planner" output="screen">
        <remap from="~goal" to="/move_base_simple/goal"/>
        <remap from="~pose" to="/airsim_node/drone_1/drone_pose"/>
    </node>

    <!-- Start local ego planner node -->
    <node pkg="navigation" type="ego_planner" name="ego_planner" output="screen">
        <remap from="~cloud" to="/airsim_node/drone_1/pointcloud/filtered"/>
        <remap from="~goal" to="/move_base_simple/goal"/>
        <remap from="~pose" to="/airsim_node/drone_1/drone_pose"/>
    </node>

    <!-- Start rviz -->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find navigation)/rviz/drone_navigation.rviz" />
</launch> 